map/bookmark_manager.cpp

```diff
@@ -1806,13 +1806,13 @@ BookmarkManager::KMLDataCollectionPtr BookmarkManager::LoadBookmarks(
   {
     auto const filePath = base::JoinPath(dir, file);
     auto kmlData = LoadKmlFile(filePath, fileType);
-    if (kmlData == nullptr)
+    if (!kmlData)
       continue;
-    if (checker && !checker(*kmlData))
+    if (checker && !checker(kmlData.value()))
       continue;
     if (m_needTeardown)
       break;
-    collection->emplace_back(filePath, std::move(kmlData));
+    collection->emplace_back(filePath, std::unique_ptr<kml::FileData>(&kmlData.value()));
```

Comment:
`collection` has type `std::vector<std::pair<std::string, std::unique_ptr<kml::FileData>>>`
I converted `std::optional<kml::FileData>` to `std::unique_ptr<kml::FileData>` here.

Reply:
Good question, likely yes, if the collection is not moved often. Moving unique_ptr is cheaper than moving FileData.

Otherwise:
```suggestion
    collection->emplace_back(filePath, std::make_unique<kml::FileData>(std::move(kmlData)));
```

Reply:
Your suggested code doesn't work. Compilation error:
`error: no matching constructor for initialization of 'kml::FileData'`

Replaced with:
`std::make_unique<kml::FileData>(std::move(kmlData.value()))`